x-pack/metricbeat/module/azure/client.go

```diff
@@ -105,29 +107,30 @@ func (client *Client) GetMetricValues(report mb.ReporterV2) error {
 			}
 			filter = strings.Join(filterList, " AND ")
 		}
-		resp, err := client.AzureMonitorService.GetMetricValues(metric.Resource.ID, metric.Namespace, metric.TimeGrain, timespan, metric.Names,
+		resp, err := client.AzureMonitorService.GetMetricValues(metric.Resource.SubID, metric.Namespace, metric.TimeGrain, timespan, metric.Names,
 			metric.Aggregations, filter)
 		if err != nil {
-			err = errors.Wrapf(err, "error while listing metric values by resource ID %s and namespace  %s", metric.Resource.ID, metric.Namespace)
-			client.LogError(report, err)
+			err = errors.Wrapf(err, "error while listing metric values by resource ID %s and namespace  %s", metric.Resource.SubID, metric.Namespace)
+			client.Log.Error(err)
+			report.Error(err)
```

Comment:
nit: maybe add return nil here, then you don't need the `else{}` part. Might make the code easier to read.

Reply:
@kaiyan-sheng , there is a loop at the top of the method, if there is an error we report it, else we map the metric values and we continue in the loop. We return only the list at the end. Let me know if more clarification is necessary.

Reply:
ahh didn't see the for loop, thanks!